mdc_weight_mgmt_xml_import/models/maintenance_equipment.py

```diff
@@ -0,0 +1,174 @@
+                        'error_message': (_("Processing failed: %s",e)),
+                        'traceback': traceback.format_exc(),
+                    })
+
+        if errors:
+            error_model = self.env['mdc.weight.record.error'].sudo()
+            created_errors = error_model.create(errors)
+            created_errors.action_send_mail()
+
+    def _process_xml_file(self, file, processed_path, record):
+        logger = logging.getLogger(__name__)
+        logger.info("Processing file: %s", file.name)
+
+        tree = ET.parse(file)
+        root = tree.getroot()
+        nsmap = root.nsmap
+
+        product_code = self.get_statvalue_by_id('103', root=root, nsmap=nsmap)
+        machine_text = self.get_text_from('Machine_Number', root=root, nsmap=nsmap)
+        start_str = self.get_text_from('Production_Start', root=root, nsmap=nsmap)
+        end_str = self.get_text_from('Production_End', root=root, nsmap=nsmap)
+        period_str = self.get_statvalue_by_id('11', root=root, nsmap=nsmap)
+        weight_nom = self.get_statvalue_by_id('21', root=root, nsmap=nsmap)
+        weight_dec = self.get_statvalue_by_id('1262', root=root, nsmap=nsmap)
+        unit_ok = self.get_statvalue_by_id('30', unit='6', root=root, nsmap=nsmap)
+        weight_ok_tot_qty = self.get_statvalue_by_id('30', unit='4', root=root, nsmap=nsmap)
+        unit_reject_low = self.get_statvalue_by_id('34', unit='6', root=root, nsmap=nsmap)
+        unit_reject_exceed = self.get_statvalue_by_id('36', unit='6', root=root, nsmap=nsmap)
+
+        fmt = "%Y-%m-%d %H:%M"
+        start_dt = datetime.strptime(start_str, fmt) if start_str else None
+        end_dt = datetime.strptime(end_str, fmt) if end_str else None
+        t= datetime.strptime(period_str, "%H:%M:%S")
+        period_min = t.hour * 60 + t.minute + t.second / 60
+        machine_number = re.match(r'\d+', machine_text).group() if machine_text else None
+        equipment = self.env['maintenance.equipment'].search([('serial_no', '=', machine_number)], limit=1)
+        if not equipment or equipment.id != record.id:
+            raise Exception(_(
+                "Equipment mismatch or not found.\nExpected: %(expected)s\nFound: %(found)s",
+                expected=record.name,
+                found=equipment.name if equipment else "None"
+            ))
+        product = self.env['product.product'].search([('default_code', '=', product_code)], limit=1)
+        if not product or not product_code:
+            raise Exception(_("Product with code %(code)s not found", code=product_code))
+
+        duplicate = self.env['mdc.weight.record'].search([
+            ('equipment_id', '=', equipment.id),
+            '|',
+                '&',
+                    ('start', '<=', start_dt),
+                    ('end', '>=', start_dt),
+                '&',
+                    ('start', '<=', end_dt),
+                    ('end', '>=', end_dt),
+        ], limit=1)
```

Comment:
Please move this to `create` method for `mdc.weight.record`. Or even better, to an `@api.contrains` method, that is the more centralized option. If this code is kept only here, no error should be raised if we create a record at other point, like record creation or update via `form` or other.